### DISCUSSION TITLE
ArgumentCountError: Too few arguments to function App\Models\MyModel:scopeRole(), 1 passed in

### DISCUSSION THREAD

**Discussion by ubudab109** [spatie/laravel-permission#2020] [Q&A]
When I want to filter users based on the role name, an error like this appears. Here is the query I use to filter data by role.

> ArgumentCountError: Too few arguments to function App\Models\MyModel::scopeRole(), 1 passed in

`$this->model`
`->when($role != '' || $role != null, function ($query) use ($role) {
        $query->whereHas('role', function ($roles) use ($role) {
          $roles->where('name','LIKE','%'.$role.'%');
        })`

**Comment by erikn69** [comment]
maybe you are overwriting `scopeRole`
https://github.com/spatie/laravel-permission/blob/6a3ed627cee28a552b5176c172ae0abc5eb30925/src/Traits/HasRoles.php#L73

**Reply by ubudab109** [reply]
> maybe you are overwriting `scopeRole`
> 
> https://github.com/spatie/laravel-permission/blob/6a3ed627cee28a552b5176c172ae0abc5eb30925/src/Traits/HasRoles.php#L73

Thanks for your reply, but i didn't overwrite any file in laravel permission package

**Reply by erikn69** [reply]
>i didn't overwrite any file in laravel permission package

I didn't say overwrite files in laravel permission package, i did say `scopeRole` on user model from your implementation
[must-not-have-a-role-or-roles-property-nor-a-roles-method](https://spatie.be/docs/laravel-permission/v5/prerequisites#must-not-have-a-role-or-roles-property-nor-a-roles-method)
